src/Marten/QuerySession.cs

```diff
@@ -119,13 +119,10 @@ private T load<T>(object id)
         private void assertCorrectIdType<T>(object id)
         {
             var mapping = Tenant.MappingFor(typeof(T));
-            if (id.GetType() != mapping.IdType)
+            if (id.GetType() != mapping.IdType && (id.GetType() != typeof(int) || mapping.IdType != typeof(long)))
```

Comment:
Adding this ` && (id.GetType() != typeof(int) || mapping.IdType != typeof(long)` in the parent `if` block looks to be changing the behaviour of earlier implementation. As per my understanding of earlier logic, if `id.GetType()` and `mapping.IdType` did not match, the subsequent check inside `if (id.GetType() == typeof(int) && mapping.IdType == typeof(long))` means that `int` and `long` are still compatible hence exception is not required to be thrown.

Can you please let me know your thoughts on the above?

Reply:
@mysticmind I just simplified the previous condition.

So I at first inverted the condition from line https://github.com/JasperFx/marten/pull/1428/files#diff-2f5f5939e30dd61deec7ee3492372bc9L124.

Then it was:
```
if (id.GetType() != mapping.IdType)
{
    if (id.GetType() != typeof(int) || mapping.IdType != typeof(long))
        throw new InvalidOperationException(
            $"The id type for {typeof(T).FullName} is {mapping.IdType.Name}, but got {id.GetType().Name}");
}
```
having that I joined two conditions to the final form

Reply:
The code in master is as below:
https://github.com/JasperFx/marten/blob/master/src/Marten/QuerySession.cs#L119|
```
        private void assertCorrectIdType<T>(object id)
        {
            var mapping = Tenant.MappingFor(typeof(T));
            if (id.GetType() != mapping.IdType)
            {
                if (id.GetType() == typeof(int) && mapping.IdType == typeof(long))
                    return;

                throw new InvalidOperationException(
                    $"The id type for {typeof(T).FullName} is {mapping.IdType.Name}, but got {id.GetType().Name}");
            }
        }
```

Reply:
Reverted the change - I agree my version was too confusing,

Reply:
@oskardudycz  with ref to our DM. I was bit sleep deprived. The changes result in equivalent logic but the earlier logic was bit explicit on why we are running those checks. Thanks for reverting it.